src/execution/ingestion_sample.cpp

```diff
@@ -0,0 +1,680 @@
+#include "duckdb/common/types/data_chunk.hpp"
+#include "duckdb/common/vector_operations/vector_operations.hpp"
+#include "duckdb/execution/reservoir_sample.hpp"
+
+namespace duckdb {
+
+idx_t IngestionSample::NumSamplesCollected() const {
+	if (!sample_chunk) {
+		return 0;
+	}
+	return sample_chunk->size();
+}
+
+idx_t IngestionSample::NumActiveSamples() const {
+	auto ret = MaxValue<idx_t>(actual_sample_indexes.size(), base_reservoir_sample->reservoir_weights.size());
+	D_ASSERT(ret <= FIXED_SAMPLE_SIZE);
+	return ret;
+}
+
+unique_ptr<DataChunk> IngestionSample::GetChunk(idx_t offset) {
+	throw InternalException("Invalid Call to Get Chunk");
+}
+
+unique_ptr<DataChunk> IngestionSample::GetChunkAndShrink() {
+	throw InternalException("Invalid Sampling state");
+}
+
+unique_ptr<DataChunk> IngestionSample::CreateNewSampleChunk(vector<LogicalType> &types, idx_t size) const {
+	auto new_sample_chunk = make_uniq<DataChunk>();
+	new_sample_chunk->Initialize(Allocator::DefaultAllocator(), types, size);
```

Comment:
Can we leave out the columns we are not going to sample entirely from this chunk?